chirp/src/chirp_swig_wrap.c

```diff
@@ -54,6 +54,18 @@ char *chirp_wrap_whoami(const char *hostname, time_t stoptime)
 	return xxstrdup(id);
 }
 
+char *chirp_wrap_hash(const char *hostname, const char *path, const char *algorithm, time_t stoptime) {
+	int result;
+	unsigned char digest[CHIRP_DIGEST_MAX];
+
+	result = chirp_reli_hash(hostname, path, algorithm, digest, stoptime);
+
+	if(result < 0)
+		return NULL;
+
+	return xxstrdup( (char *) digest);
+}
+
```

Comment:
The digest is in binary and variable size (the digest size is result if > 0). So we can't use xxstrdup.